07-routing-gossip.md

```diff
@@ -798,6 +897,42 @@ and allows bootstrapping for new nodes as well as updating for nodes that
 have been offline for some time.  The `gossip_queries` option
 allows for more refined synchronization.
 
+### Inventory-based Rebroadcasting
+
+With the simple rebroadcasting scheme described above, nodes that are connected to many different peers will receive duplicates for the same gossip messages.
+
+A more efficient scheme based on inventory messages, similar to how transactions and blocks are propagated on the bitcoin p2p network, can optionally be used: if the `option_inv_gossip` feature bit is set, instead of broadcasting `channel_annoucement` and `channel_update` messages, nodes will broadcast a `gossip_inventory` message which is similar to a `reply_channel_range` message and can be used to filter announcements by timestamp or by content (checksum):
+
+1. type: 266 (`gossip_inventory`)
+2. data:
+    * [`32`:`chain_hash`]
+	* [`tlvs`:`gossip_inventory_tlvs`]
+
+1. tlvs: `gossip_inventory_tlvs`
+2. types:
+   1. type: 1 (`channel_inv`)
+   2. data:
+      * [`tlv_len`:`encoded_short_ids`]
+   1. type: 3 (`timestamps`)
+   2. data:
+	  * [`1`:`encoding_type`]
+	  * [`tlv_len-1`:`encoded_timestamps`]
+   1. type: 5 (`checksums`)
+   2. data:
+        * [`tlv_len`:`encoded_checksums`]
+   1. type: 7 (`node_inv`)
+   2. data:
+      * [`tlv_len`:`node_ids`]
```

Comment:
from my understanding, #557 doesn't have a way of querying for `node_announcement` without the corresponding `short_channel_id`. Can you explain how one would retrieve the `short_channel_id` from this message to use in the query? If `node_ids` is a different size from `encoded_short_ids`, which one should the receiver use?

Reply:
That's a good point and now I see why @Roasbeef wanted a different INV message for node announcements. A simple answer would be to include `node_id`, `timestamp` and `short_channel_id` in the INV message, receiver can then use the provided `short_channel_id` to query the node announcement.